pkg/chains/signing/kms/kms.go

```diff
@@ -120,6 +137,28 @@ func NewSigner(ctx context.Context, cfg config.KMSSigner) (*Signer, error) {
 	}, nil
 }
 
+// getKMSAuthToken retreives token from the given mount path
+func getKMSAuthToken(dir string) (string, error) {
+	tokenEnv := "KMS_AUTH_TOKEN"
+	stat, err := os.Stat(dir)
```

Comment:
I wonder if it's necessary to check that the path exists and it is a directory. We could simply do

```go
filePath := filepath.Join(dir, tokenEnv)
fileData, err := os.ReadFile(filePath)
```

which should catch those cases and more.

Reply:
Thanks for this comment. removed check for directory with os.Stat and tested. I may be helpful to user to discover that the mountpath is incorrect  but with a detailed error message the user should be able to figure out the issue. Hence removed as suggested.